lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -811,7 +811,18 @@ def create_new_key_pair(region, pair_name)
       ssh_string = public_key()
       region.key_pairs.import(pair_name, ssh_string)
       kp = region.key_pairs[pair_name]
-      if kp.exists?
+
+      exists = false
+      for tries in 1..10
```

Comment:
As far as how long it takes when it needs to wait: we've only seen this intermittently in CI, so I haven't actually seen it happen with this code.